File: solution/110.balanced-binary-tree/balanced-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int height(struct TreeNode * root){
    if(!root){
        return 0;
    }
    int left = height(root->left) + 1;
    int right = height(root->right) + 1;
    return left > right ? left : right;
    
}
bool isBalanced(struct TreeNode* root) {
    if(!root){
        return 1;
    }
    int leftTree = isBalanced(root->left);    
    int rightTree = isBalanced(root->right);
    int left = height(root->left);    
    int right = height(root->right);
    int temp = right - left;
    if(temp <=1 && temp >= -1){
        return leftTree && rightTree;
    }else {
        return 0;
    }

}

File: solution/110.balanced-binary-tree/balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
function height(root) {
        if (!root) {
          return 0;
        }
        let left = height(root.left) + 1;
        let right = height(root.right) + 1;
        return left > right ? left : right;
      }
      var isBalanced = function(root) {
        if (!root) {
          return 1;
        }
        let leftTree = isBalanced(root.left);
        let rightTree = isBalanced(root.right);
        let left = height(root.left);
        let right = height(root.right);
        let temp = right - left;
        if (temp <= 1 && temp >= -1) {
          return leftTree && rightTree;
        } else {
          return 0;
        }
      };